liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/applications/DeployMojo.java

```diff
@@ -165,22 +179,35 @@ private void installDependencies() throws Exception {
     protected void installProject() throws Exception {
         if (isSupportedType(project.getPackaging())) {
             if (looseApplication) {
-                installLooseApplication(project);
+                //Either write loose app xml file to buildDir and deploy, or install to serverDir
+                if (shouldDeploy()) {
+                    appArchive = new File(project.getBuild().getDirectory(), getLooseConfigFileName(project));
+                    //Generates looseAppFile in the project's buildDir
+                    installLooseApplication(project, appArchive);
+                    //Loose app xml deployed to server
+                    deployApp();
+                } else {
+                    //Writes loose app xml straight to serverDir
+                    installLooseApplication(project, new File(new File(serverDirectory, getAppsDirectory()), getLooseConfigFileName(project)));
+                }
             } else {
-                installApp(project.getArtifact());
+                if (shouldDeploy()) {
+                    deployApp();
+                } else {
+                    installApp(project.getArtifact());
+                }
             }
         } else {
             throw new MojoExecutionException(
                     MessageFormat.format(messages.getString("error.application.not.supported"), project.getId()));
         }
     }
 
-    private void installLooseApplication(MavenProject proj) throws Exception {
+    private void installLooseApplication(MavenProject proj, File looseConfigFile) throws Exception {
```

Comment:
The first couple lines of code are getting the looseConfigFileName and application name from the proj. Shouldn't it be using the passed in looseConfigFile parameter for that?

Reply:
Yup. Fixed.